service/r2/mocks/storage_mock.go

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```

Comment:
Can you set up the build script so the mocks can be auto-generated via mock-gen? This is pretty standard practice in other repos btw, so next time you change the `Store` interface, you just need to run `make mock-gen`.